heft-plugins/heft-webpack4-plugin/src/shared.ts

```diff
@@ -23,25 +23,40 @@ export type IWebpackConfiguration =
 /**
  * @public
  */
-export interface IWebpackBundleSubstageProperties extends IBundleSubstageProperties {
+export interface IWebpackVersions {
+  webpackVersion: string;
+  webpackDevServerVersion: string;
+}
+
+/**
+ * @public
+ */
+export interface IWebpackPluginAccessor {
   /**
-   * The configuration used by the Webpack plugin. This must be populated
-   * for Webpack to run. If webpackConfigFilePath is specified,
-   * this will be populated automatically with the exports of the
-   * config file referenced in that property.
+   * A hook that provides the versions of Webpack and Webpack Dev Server.
+   */
+  readonly onEmitWebpackVersionsHook?: AsyncParallelHook<IWebpackVersions>;
+  /**
+   * A hook that allows for modification of the  configuration used by the Webpack
+   * plugin. This must be populated for Webpack to run. If webpackConfigFilePath is
+   * specified, this will be populated automatically with the exports of the config
+   * file referenced in that property.
    *
    * @remarks
    * Tapable event handlers can return `null` instead of `undefined` to suppress
    * other handlers from creating a configuration object.
    */
   // We are inheriting this problem from Tapable's API
   // eslint-disable-next-line @rushstack/no-new-null
-  webpackConfiguration?: webpack.Configuration | webpack.Configuration[] | null;
-}
-
-/**
- * @public
- */
-export interface IWebpackBuildStageProperties extends IBuildStageProperties {
-  webpackStats?: webpack.Stats | webpack.compilation.MultiStats;
+  readonly onConfigureWebpackHook?: AsyncSeriesWaterfallHook<IWebpackConfiguration | null>;
+  /**
+   * A hook that provides the finalized configuration that will be used by Webpack.
+   */
+  // We are inheriting this problem from Tapable's API
+  // eslint-disable-next-line @rushstack/no-new-null
```

Comment:
You could convert any falsy value returned by `onConfigureWebpackHook` into `undefined` and pass that as input to `onAfterConfigureWebpackHook`, since it doesn't have the ability to modify the value.